src/Elcodi/Component/Core/Tests/UnitTest/Entity/AbstractEntityTest.php

```diff
@@ -0,0 +1,312 @@
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Component\Core\Tests\UnitTest\Entity;
+
+use Exception;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Class AbstractEntityTest.
+ */
+abstract class AbstractEntityTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var int
+     *
+     * Getters and setters
+     */
+    const GETTER_SETTER = 1;
+
+    /**
+     * @var int
+     *
+     * Adder and Remover
+     */
+    const ADDER_REMOVER = 2;
+
+    /**
+     * Return the entity namespace.
+     *
+     * @return string Entity namespace
+     */
+    abstract public function getEntityNamespace();
+
+    /**
+     * Return the fields to test in entities.
+     *
+     * [
+     *      [
+     *          "type" => $this::GETTER_SETTER,
+     *          "getter" => "getValue",
+     *          "setter" => "setValue",
+     *          "value" => "Elcodi\Component\...\Interfaces\AnInterface"
+     *          "nullable" => true
+     *      ],
+     *      [
+     *          "type" => $this::ADDER_REMOVER|$this::ADDER_REMOVER,
+     *          "getter" => "getValue",
+     *          "setter" => "setValue",
+     *          "adder" => "addValue",
+     *          "removed" => "removeValue",
+     *          "bag" => "collection", // can be array
+     *          "value" => "Elcodi\Component\...\Interfaces\AnInterface"
+     *      ]
+     * ]
+     *
+     * @return array Fields
+     */
+    abstract public function getTestableFields();
+
+    /**
+     * Test all desired tests.
+     */
+    public function testFields()
+    {
+        $fields = $this->getTestableFields();
+        $fields = $this->addTraits($fields);
+
+        foreach ($fields as $field) {
+            if ($field['type'] & $this::GETTER_SETTER) {
+                $this->checkGetterSetter($field);
+            }
+
+            if ($field['type'] & $this::ADDER_REMOVER) {
+                $this->checkAdderRemover($field);
+            }
+        }
+    }
+
+    /**
+     * Test getter and setter.
+     *
+     * @param array $field Field
+     */
+    private function checkGetterSetter(array $field)
+    {
+        $classNamespace = $this->getEntityNamespace();
+        $class = $this->getMock($classNamespace, null, [], '', false);
+
+        $setter = $field['setter'];
+        $getter = $field['getter'];
+        $value = $this->getFieldValue($field);
+
+        $this->assertSame(
+            $class,
+            $class->$setter($value)
+        );
+
+        $this->assertEquals(
+            $value,
+            $class->$getter()
+        );
+
+        try {
+            $class->$setter(null);
+            $this->assertNull($class->$getter());
+        } catch (Exception $e) {
+            if ($field['nullable']) {
+                $this->fail("Method $setter of entity $classNamespace is not nullable");
```

Comment:
What about a `sprintf('Method %s of entity %s is not nullable', $setter, $classNamespace)` here?